src/components/Menus/DropdownMenu.js

```diff
@@ -0,0 +1,207 @@
+import React             from 'react'
+import PropTypes         from 'prop-types'
+import Radium            from 'radium'
+import Menu              from './Menu'
+
+const styles = {
+  menuContainer: {
+    zIndex: 9999,
+    position: 'absolute',
+    display: 'none',
+    width: '100%'
+  },
+  menuContainerOpen: {
+    display: 'block'
+  }
+}
+
+@Radium
+class DropdownMenu extends React.Component {
+  static propTypes = {
+    /** MenuItems or Divider */
+    children: PropTypes.node.isRequired,
+    /** React component or HTML that will be used as the dropdown trigger. ie. Button or Icon */
+    triggerElement: PropTypes.element,
+    /** Open and close menu manually */
+    open: PropTypes.bool,
+    /** Callback fired on dropdown close */
+    onClose: PropTypes.func,
+    /** Callback fired on click */
+    onClick: PropTypes.func,
+    /** Callback fired on dropdown open */
+    onOpen: PropTypes.func,
+    /** Callback function fired on key down */
+    onKeyDown: PropTypes.func,
+    /** Requesting to open/close for controlled open component. */
+    onRequestChange: PropTypes.func,
+    /** Callback function fired when a MenuItem is selected */
+    onSelect: PropTypes.func,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({}),
+    /** Props passed to Menu component */
+    menuProps: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    open: null
+  }
+
+  state = {
+    open: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open !== null && this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.open()
+      } else {
+        this.close()
+      }
+    }
+  }
+
+  _handleSelect = (option) => {
+    const { onSelect, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(false)
```

Comment:
What is the use case for `onRequestChange`?

Reply:
`onRequestChange` is to manually control the open/close of the dropdown

Reply:
Why isn't that handle via `handleSelect`?

Reply:
`handleSelect` default behavior is to close when a MenuItem is selected. However, if you want to control the open and close manually via props that is also possible. See `Select`.

Reply:
How does `onRequestChange` play into that? Seems like you could still do that with just a `handleSelect` prop and an `open` prop.